ships/0036-runAs-for-supporting-steps.md

```diff
@@ -0,0 +1,210 @@
+
+#### Story 2
+
+As a Shipwright Build Strategy author, I want to read about recommendations to specify runAsUser and runAsGroup so that I correctly define the steps.
+
+## Proposal
+
+### API changes
+
+The Build Strategy API is extended with a top-level `securityContext` inside the `spec`. It can be used to define the `runAsUser` and `runAsGroup`. If present, those settings override the `runAsUser` and `runAsGroup` form the container templates in the Shipwright configuration. They are also the default for the strategy steps. Strategy steps can still overwrite those two fields in their step-level `securityContext`.
+
+In the following example, all Shipwright-managed steps will run as user 1001 and group 1000 while the single build strategy step will run as user 1001 and group 0.
+
+```yaml
+apiVersion: shipwright.io/v1alpha1
+kind: ClusterBuildStrategy
+metadata:
+  name: buildpacks-v3
+spec:
+  securityContext:
+    runAsUser: 1001
+    runAsGroup: 1000
+  buildSteps:
+  - name: build
+    securityContext:
+      runAsGroup: 0
+    ...
+```
+
+### Base image changes
+
+Container images define which users and groups exist in the container by specifying the content of `/etc/passwd` and `/etc/group`. Though, it is generally possible to run a container with as a user or group that is not included in the container image.
+
+One change that we must make, is to introduce a common world-writable home directory that we call `/shared-home`. We will reference this in the Shipwright configuration's container template.
+
+Unfortunately, binaries may rely on an entry in the `/etc/passwd` and `/etc/group` files to be present to properly work. For the binaries that we run in supporting steps, this is the case for `ssh` which we need in the Git step when cloning using the SSH protocol. [SSH requires an entry and fails if it is not present.](https://github.com/openssh/openssh-portable/blob/V_9_3_P1/ssh.c#L671-L676)
+
+We therefore must ensure that those files contain proper information.
+
+We will change the base image to contain `/etc/passwd` and `/etc/group` files with the content that applies to the default Shipwright configuration. That will be for `/etc/passwd`:
+
+```txt
+shp:x:1000:1000:shp:/shared-home:/sbin/nologin
+```
+
+And for `/etc/group`:
+
+```txt
+shp:x:1000
+```
+
+### BuildRun reconciler changes
+
+The BuildRun reconciler is changed to check the strategy-level `securityContext`. If it is defined, then their `runAsUser` and `runAsGroup` override those settings from the container templates of Shipwright-managed steps in the configuration. They are also defaults for the strategy steps' securityContext.
+
+If the strategy-level `securityContext` is defined, then the reconciler will setup the `/etc/passwd` and `/etc/group` files to match this configuration. The following annotations are added to the TaskRun (assuming a build strategy that runs as user 1010 and group 1020):
```

Comment:
Again, I think we need to be clear here on which containers we are modifying `/etc/passwd` and `/etc/group`. If these are for the Shipwright pre/post steps, this is fine. For arbitrary build strategy steps provided by users, this almost certainly will break things.